src/androidMain/kotlin/software/momento/kotlin/sdk/internal/InternalDataClient.android.kt

```diff
@@ -0,0 +1,91 @@
+package software.momento.kotlin.sdk.internal
+
+import com.google.protobuf.ByteString
+import grpc.cache_client.ECacheResult
+import grpc.cache_client._GetRequest
+import grpc.cache_client._GetResponse
+import grpc.cache_client._SetRequest
+import io.grpc.Metadata
+import software.momento.kotlin.sdk.auth.CredentialProvider
+import software.momento.kotlin.sdk.config.Configuration
+import software.momento.kotlin.sdk.exceptions.CacheServiceExceptionMapper
+import software.momento.kotlin.sdk.exceptions.InternalServerException
+import software.momento.kotlin.sdk.responses.cache.GetResponse
+import software.momento.kotlin.sdk.responses.cache.SetResponse
+import java.io.Closeable
+
+internal actual class InternalDataClient actual constructor(
+    credentialProvider: CredentialProvider, configuration: Configuration
+): Closeable {
+
+    private val stubsManager: DataGrpcStubsManager
+
+    init {
+        stubsManager = DataGrpcStubsManager(credentialProvider)
+    }
+
+    internal actual suspend fun set(
+        cacheName: String, key: String, value: String, ttlMilliSeconds: Long
```

Comment:
We'll need to do validation on all of this. I can add it in a second pass if you'd like.